packages/tools/src/utilities/math/polyline/combinePolyline.ts

```diff
@@ -0,0 +1,327 @@
+  targetPolyline: Types.Point2[],
+  sourcePolyline: Types.Point2[]
+) {
+  const targetPolylinePoints: PolylinePoint[] = [];
+  const sourcePolylinePoints: PolylinePoint[] = [];
+  const sourceIntersectionsCache = new Map<
+    number,
+    PolylineIntersectionPoint[]
+  >();
+
+  let intersectionPointDirection = PolylinePointDirection.Entering;
+
+  // Store all vertices and intersection for target contour
+  for (let i = 0, len = targetPolyline.length; i < len; i++) {
+    const p1 = targetPolyline[i];
+    const pointInside = containsPoint(sourcePolyline, p1);
+    const vertexPoint: PolylinePoint = {
+      type: PolylinePointType.Vertex,
+      coordinates: p1,
+      position: pointInside
+        ? PolylinePointPosition.Inside
+        : PolylinePointPosition.Outside,
+      visited: false,
+      next: null,
+    };
+
+    targetPolylinePoints.push(vertexPoint);
+
+    const q1 = targetPolyline[i === len - 1 ? 0 : i + 1];
+    const intersectionsInfo = getLineSegmentIntersectionsIndexes(
+      sourcePolyline,
+      p1,
+      q1
+    ).map((intersectedLineSegment) => {
+      const sourceLineSegmentId: number = intersectedLineSegment[0];
+      const p2 = sourcePolyline[intersectedLineSegment[0]];
+      const q2 = sourcePolyline[intersectedLineSegment[1]];
+
+      // lineSegment.intersectLine returns `undefined` when the intersection
+      // is at one of the line vertices.
+      // Examples:
+      //   - [(0, 0), (1, 1)] x [(1, 1), (1, 2)]
+      //   - [(0, 1), (2, 1)] x [(1, 1), (1, 2)]
+      const intersectionCoordinate = getLineSegmentsIntersection(
+        p1,
+        q1,
+        p2,
+        q2
+      ) as Types.Point2;
+
+      const targetStartPointDistSquared = math.point.distanceToPointSquared(
+        p1,
+        intersectionCoordinate
+      );
+
+      return {
+        sourceLineSegmentId,
+        coordinate: intersectionCoordinate,
+        targetStartPointDistSquared,
+      };
+    });
+
+    intersectionsInfo.sort(
+      (left, right) =>
+        left.targetStartPointDistSquared - right.targetStartPointDistSquared
+    );
+
+    intersectionsInfo.forEach((intersectionInfo) => {
+      const { sourceLineSegmentId, coordinate: intersectionCoordinate } =
+        intersectionInfo;
+
+      // Intersection point to be added to the target polyline list
+      const targetEdgePoint: PolylineIntersectionPoint = {
+        type: PolylinePointType.Intersection,
+        coordinates: intersectionCoordinate,
+        position: PolylinePointPosition.Edge,
+        direction: intersectionPointDirection,
+        visited: false,
+        next: null,
+      };
+
+      // Intersection point to be added to the source polyline list.
+      // At this point there is no way to know if the point is entering or
+      // exiting the intersection region but that is not going to be used
+      // hence it is set to "unknown".
+      const sourceEdgePoint: PolylineIntersectionPoint = {
+        ...targetEdgePoint,
```

Comment:
structuredClone is faster than spread 

https://www.measurethat.net/Benchmarks/Show/17968/0/structuredclone-vs-spread-object

Reply:
![structuredClone_vs_spread](https://github.com/cornerstonejs/cornerstone3D/assets/931820/4e201539-36f8-42ac-9266-a53df11188ec)